tests/jobs/test_triggerer_job.py

```diff
@@ -787,3 +801,323 @@ def non_pytest_handlers(val):
     assert qh.__class__ == LocalQueueHandler
     assert qh.queue == listener.queue
     listener.stop()
+
+
+class RemoteJob:
+    RUNNING_STATUS = "running"
+    FAILED_STATUS = "failed"
+    SUCCESS_STATUS = "success"
+
+    def __init__(self, status):
+        self.status = status
+
+    def get_status(self):
+        return self.status
+
+    def set_status(self, status):
+        self.status = status
+
+    def kill(self):
+        self.set_status(RemoteJob.FAILED_STATUS)
+
+
+class RemoteService:
```

Comment:
These singleton classes (this one and the one below) are a bit overkill IMO. They can just be regular classes and instantiate regular test-scoped objects. It’d be much easier to debug too.

Reply:
The purpose of this approach is that the trigger, once instantiated, isn't stored anywhere and is directly put into running state after initialization, making it impossible to access the instance directly. Therefore, I have made it register itself in a singleton during initialization, which allows me to access it